GUI/CloneFakeKspDialog.cs

```diff
@@ -0,0 +1,243 @@
+                return;
+            }
+            if (textBoxNewPath.TextLength == 0)
+            {
+                user.RaiseError("Please enter a path for the new instance.");
+                return;
+            }
+
+            string newName = textBoxNewName.Text;
+            string newPath = textBoxNewPath.Text;
+
+            // Clone the specified instance.
+            // Done in a new task to not block the GUI thread.
+            if (radioButtonClone.Checked)
+            {
+                try
+                {
+                    Task.Run(() => {
+                        user.RaiseMessage("Cloning instance...");
+
+                        manager.CloneInstance(selectedInstance, newName, newPath);
+
+                        if (checkBoxSetAsDefault.Checked)
+                        {
+                            manager.SetAutoStart(newName);
+                        }
+
+                        if (checkBoxSwitchInstance.Checked)
+                        {
+                            manager.SetCurrentInstance(newName);
+                        }
+
+                        // Update the list in the ChooseKSPInstanceForm.
+                        Util.Invoke(chooseKSPInstanceForm, chooseKSPInstanceForm.UpdateInstancesList);
+
+                        user.RaiseMessage("Successfully cloned instance.");
+                    });
+                }
+                catch (NotKSPDirKraken)
+                {
+                    user.RaiseError("The instance you wanted to clone is not valid.");
+                    return;
+                }
+                catch (IOException exception)
+                {
+                    user.RaiseError("The destination folder is not empty or invalid: " + exception.Message);
+                    return;
+                }
+                           
+            }
+            // Create a new dummy instance.
+            // Also in a separate task.
+            else if (radioButtonFake.Checked)
+            {
+                Versioning.KspVersion kspVersion = Versioning.KspVersion.Parse(comboBoxKspVersion.Text);
+                string dlcVersion = textBoxDlcVersion.Text;
+
+                try
+                {
+                    Task.Run(() =>
+                    {
+                        user.RaiseMessage("Creating new instance...");
+
+                        manager.FakeInstance(newName, newPath, kspVersion, dlcVersion);
+
+                        if (checkBoxSetAsDefault.Checked)
+                        {
+                            manager.SetAutoStart(newName);
+                        }
+
+                        if (checkBoxSwitchInstance.Checked)
+                        {
+                            manager.SetCurrentInstance(newName);
+                        }
+
+                        // Update the list in the ChooseKSPInstanceForm.
+                        Util.Invoke(chooseKSPInstanceForm, chooseKSPInstanceForm.UpdateInstancesList);
```

Comment:
Regarding the question of how to update the instance list after the form closes, what about calling `UpdateInstancesList` in `CloneFakeInstanceMenuItem_Click`? That's how the other buttons on the instance selection form work.

Reply:
Well, I tried to clone my main KSP instance as a test, and copying 4 GB of data takes some time. This completely blocked the GUI.

Reply:
Another solution I see is to leave the clone/fake dialog open until the process is done, with a process bar or something to let the user know there's something happening in the background.

Reply:
Yeah, I think a progress bar would be good.